AC-906 : Re-Designed Home Screen UI

sahuadarsh0 opened the pull request:
## Description of what I changed
Home Screen UI is re-designed both for dark mode and light mode

## Issue I worked on
<!--- This project only accepts pull requests related to open issues -->
<!--- Want a new feature or change? Discuss it in an issue first -->
<!--- Found a bug? Point us to the issue/or create one so we can reproduce it -->
<!--- Just add the issue number at the end: -->
JIRA Issue: https://issues.openmrs.org/browse/AC-906

## Checklist: I completed these to help reviewers :)
- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1). 

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

## ScreenShots
### OLD

![old](https://user-images.githubusercontent.com/25864174/111957619-44826300-8b12-11eb-929d-ea8e9ed20a15.png)



### NEW
Device Nexus 4 API 21 

![Screenshot_1616399073](https://user-images.githubusercontent.com/25864174/111956807-38e26c80-8b11-11eb-87c9-730e4a612f1e.png) ![Screenshot_1616399081](https://user-images.githubusercontent.com/25864174/111956810-3a139980-8b11-11eb-9975-ad4b598ac626.png)

Device Mi A3 API 29
![photo_2021-03-21_19-56-37](https://user-images.githubusercontent.com/25864174/111957056-8959ca00-8b11-11eb-9f24-50dcf83a4a83.jpg) ![photo_2021-03-21_19-56-34](https://user-images.githubusercontent.com/25864174/111957073-8fe84180-8b11-11eb-9c16-04a1a6aaf007.jpg)

LuGO0 commented:
did we remove the bottom bar ??

LuGO0 commented:
![1](https://user-images.githubusercontent.com/30355336/111997400-0486a480-8b41-11eb-9daa-2ab7b4bf2fcc.png)

ShivanshGoel221B commented:
> Just some thoughts on the design,
> 1. did we remove the bottom navigation bar? that was looking good I think we should use it.
> 2. I think we should remove the long `OpenMRS Android client` to just `OpenMRS` looks better.
> 3. I dont know what looks better with welcome or without it? I think we need a poll here 😄.
> 4. you can fix the dimensions of the logo in the title bar as well looks bigger to me (maybe change it to the logo you built in the original mock).

I think it would look better without that 'Welcome, ' there.

sahuadarsh0 commented:
@LuGO0  
1.Yes, I actually removed the bottom navigation because that can be better handled in a single activity and multiple fragments using nav graphs as recommended by Google, and here we have activity for each of the dashboard items that's why not implemented it
2. I am working on removing the Long name.
3. for now I am removing the welcome too, if better we'll add that later.
4. I am making changes in the action bar/toolbar to make it as it is in my mock

LuGO0 commented:
> @LuGO0
> 1.Yes, I actually removed the bottom navigation because that can be better handled in a single activity and multiple fragments using nav graphs as recommended by Google, and here we have activity for each of the dashboard items that's why not implemented it
> 2. I am working on removing the Long name.
> 3. for now I am removing the welcome too, if better we'll add that later.
> 4. I am making changes in the action bar/toolbar to make it as it is in my mock

I think bottom navigation can be implemented via single activity and nav graphs, you sure we cant do this?

Akshatji800 reviewed (commented):
@sahuadarsh0 I liked the suggestions suggested by @LuGO0  and the prototype shown it looks far better than the previous one but yes I want to add some more suggestions:

1. Since it is suggested to have only openMRS as the logo on the top and it looks better also but there the 2 options one of switching to different session location and the other of disconnecting from server option is excluded. You can include both the options at the right side of search bar using a side navigation drawer or you can give them a place in bottom navigation bar in place of some other options. I prefer having them arranged in bottom navigation bar only, it can arrange five options.
2. I also suggest to remove that "Welcome".

sahuadarsh0 commented:
@LuGO0 Yes we can implement it ... but first let's see what is going to final in [Home UI Redesign Discussion ](https://talk.openmrs.org/t/re-design-home-screen-with-a-better-ui/32652) 
@ShivanshGoel221B I think your idea for implementing location switching in bottom navigation is fab. I am looking forward to implementing it in the next mock. Thanks

Glock123 commented:
Can authors and contributors of this project create some issues so that newbies like me can contribute and get more familiar with the massive codebase?

LuGO0 commented:
> Can authors and contributors of this project create some issues so that newbies like me can contribute and get more familiar with the massive codebase?

we have Issues in JIRA you can take up some beginner Issues there 👍🏼

LuGO0 commented:
> @LuGO0 Yes we can implement it ... but first let's see what is going to final in [Home UI Redesign Discussion ](https://talk.openmrs.org/t/re-design-home-screen-with-a-better-ui/32652)
> @ShivanshGoel221B I think your idea for implementing location switching in bottom navigation is fab. I am looking forward to implementing it in the next mock. Thanks

@sahuadarsh0 I think the discussion there was about moving towards carbon design which we can slowly migrate to but I think we can go forward with this redesign because this is not only about color codes but UI components will be added in this redesing which would be useful for carbon system as well 👍🏼

sahuadarsh0 commented:
Hey, @LuGO0 I am stuck. What items shall I put on the bottom navigation view? 
1. Find patients 
2. Active Visits //  seems unnecessary. any alternative suggestion
3. location
4. settings

any thoughts @rishabh-997 

Or shall i continue with this.

sahuadarsh0 commented:
### New Changes I have made
1. Removed all bitmap and bind drawable resources instead used vector files for faster loading across all screen sizes. Also removed changeColorOfDashboardIcons cause icons look great in both dark and light mode
2. changed position of switching to different session location to bottom navigation as suggested by @Akshatji800 
3. Removed Welcome
4. Removed Settings and location from the action bar 

@LuGO0 @rishabh-997 Review the code for necessary changes.

Screenshot
![photo_2021-03-31_13-39-57](https://user-images.githubusercontent.com/25864174/113112573-f0b8fd80-9226-11eb-9c26-f166f62e516d.jpg)

sahuadarsh0 commented:
@rishabh-997 @LuGO0 any follow-up?

LuGO0 commented:
sorry @sahuadarsh0 for the late follow up changes look great IMO, I will wait for @rishabh-997 here which might take a bit more time.

LuGO0 commented on openmrs-client/src/main/res/menu/basic_menu.xml:
```diff
@@ -23,16 +23,16 @@
+    <!--        android:orderInCategory="100"-->
+    <!--        android:title="@string/action_settings"-->
+    <!--        app:showAsAction="never" />-->
     <item
```
do we need these commented lines @sahuadarsh0 ?

sahuadarsh0 replied:
I was testing things out and not sure which are going to be final that's why I put those comments. We don't need those commented lines. Removing it

LuGO0 commented on openmrs-client/src/main/res/menu/bottom_nav_menu.xml:
```diff
@@ -0,0 +1,38 @@
+        android:id="@+id/actionSettings"
+        android:icon="@drawable/ic_settings"
+        android:title="@string/action_settings" />
+</menu>
```
looks like you have added navigation graphs as well please mention that in the PR description.

sahuadarsh0 replied:
No, I have not added any other navigation graphs.

LuGO0 commented on openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java:
```diff
@@ -174,16 +169,18 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
+    protected void callStartActivityForResult() {
+        startActivityForResult(new Intent(this, SettingsActivity.class), ApplicationConstants.RequestCodes.START_SETTINGS_REQ_CODE);
+    }
+
```
This dosent look good to me looks like a custom logic specific for opening settingsActivity? I think there must be a more generic way to handle this !!

LuGO0 commented:
> Hey, @LuGO0 I am stuck. What items shall I put on the bottom navigation view?
> 
> 1. Find patients
> 2. Active Visits //  seems unnecessary. any alternative suggestion
> 3. location
> 4. settings
> 
> any thoughts @rishabh-997
> 
> Or shall i continue with this.

what about about us

sahuadarsh0 commented:
> what about `about us` for now as a placeholder, we can change it later to the profile dashboard as it is currently blocked due to some Issues

I thought about it before but  `about us`  doesn't seems good in the bottom navigation. As we put only those items which are frequently used and important for the app to navigate. But as later we are changing it to profile dashboard, I think using about us is a good idea. I am putting it in the end

sahuadarsh0 commented:
@LuGO0  I have fixed your requested changes but I am getting into trouble while managing conflict files that were recently merged into the master branch. can you help me out.

sahuadarsh0 commented:
Closing this PR due to lots of conflicts with the master branch and creating a [new PR](https://github.com/openmrs/openmrs-contrib-android-client/pull/889) with all the requested changes applied.
